### DISCUSSION TITLE
Is it possible to add options after calling a plot subroutine?

### DISCUSSION THREAD

**Discussion by macardosa** [kookma/ogpf#30] [Q&A]
I'm unable to see the plots with wxt term in macOS because Gnuplot exist and the window close very fast. 

Using persistent:
```
gnuplot -p ogpf_temp_script.gp
```
does not work.
I managed to see the plot by adding **pause mouse close** or **pause -1** to the end of the script.
I wonder if it is possible to set this option from the Fortran program.

Thanks in advance

Best
Manuel

**Comment by kookma** [comment]
@macardosa - would you please test a simple plot persistent directly from gnuplot?
I mean without Fortran, directly open gnuplot and see how it works on your system!

**Reply by macardosa** [reply]
Hello, thanks for the quick response.
If I run the script generated by the program as 
```
gnuplot -p ogpf_temp_script.gp
```
I see it succeed to plot it, just that the wxwidget window closes almost immediately, so I can't even look at the plot. 
Then I found that a solution for me was to add 
```
pause mouse close
```
to the end of the script. Maybe I can open in fortran the script and append this line if there were no way to add it as an option.

Best

**Reply by kookma** [reply]
> to the end of the script. Maybe I can open in fortran the script and append this line if there were no way to add it as an option.

Do it! I have not test the code on MacOS!

**Comment by kookma** [comment]
Note: when you plot something after that, you cannot pipe anything!
Actually Fortran sends all data to a text file and gnuplot reads it!
